src/pages/EnablePayments/PersonalInfo/PersonalInfo.tsx

```diff
@@ -0,0 +1,110 @@
+import * as Wallet from '@userActions/Wallet';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import INPUT_IDS from '@src/types/form/WalletAdditionalDetailsForm';
+import Address from './substeps/AddressStep';
+import Confirmation from './substeps/ConfirmationStep';
+import DateOfBirth from './substeps/DateOfBirthStep';
+import FullName from './substeps/FullNameStep';
+import PhoneNumber from './substeps/PhoneNumberStep';
+import SocialSecurityNumber from './substeps/SocialSecurityNumberStep';
+
+const PERSONAL_INFO_STEP_KEYS = INPUT_IDS.PERSONAL_INFO_STEP;
+const bodyContent: Array<React.ComponentType<SubStepProps>> = [FullName, DateOfBirth, Address, PhoneNumber, SocialSecurityNumber, Confirmation];
+
+function PersonalInfoPage() {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+
+    const [walletAdditionalDetails] = useOnyx(ONYXKEYS.WALLET_ADDITIONAL_DETAILS);
+    const [walletAdditionalDetailsDraft] = useOnyx(ONYXKEYS.FORMS.WALLET_ADDITIONAL_DETAILS_DRAFT);
+
+    const values = useMemo(() => getSubstepValues(PERSONAL_INFO_STEP_KEYS, walletAdditionalDetailsDraft, walletAdditionalDetails), [walletAdditionalDetails, walletAdditionalDetailsDraft]);
+    const submit = () => {
+        // TODO: uncomment after connecting steps https://github.com/Expensify/App/issues/36648
+        // const personalDetails = {
+        //     phoneNumber: (values.phoneNumber && parsePhoneNumber(values.phoneNumber, {regionCode: CONST.COUNTRY.US}).number?.significant) ?? '',
+        //     legalFirstName: values?.[PERSONAL_INFO_STEP_KEYS.FIRST_NAME] ?? '',
+        //     legalLastName: values?.[PERSONAL_INFO_STEP_KEYS.LAST_NAME] ?? '',
+        //     addressStreet: values?.[PERSONAL_INFO_STEP_KEYS.STREET] ?? '',
+        //     addressCity: values?.[PERSONAL_INFO_STEP_KEYS.CITY] ?? '',
+        //     addressState: values?.[PERSONAL_INFO_STEP_KEYS.STATE] ?? '',
+        //     addressZip: values?.[PERSONAL_INFO_STEP_KEYS.ZIP_CODE] ?? '',
+        //     dob: values?.[PERSONAL_INFO_STEP_KEYS.DOB] ?? '',
+        //     ssn: values?.[PERSONAL_INFO_STEP_KEYS.SSN_LAST_4] ?? '',
+        // };
+        // Attempt to set the personal details
+        // Wallet.updatePersonalDetails(personalDetails);
+        Navigation.goBack(ROUTES.SETTINGS_WALLET);
+        Wallet.resetWalletAdditionalDetailsDraft();
+    };
+
+    const startFrom = useMemo(() => getInitialSubstepForPersonalInfo(values), [values]);
+
+    const {
+        componentToRender: SubStep,
+        isEditing,
+        nextScreen,
+        prevScreen,
+        moveTo,
+        screenIndex,
+    } = useSubStep({
+        bodyContent,
+        startFrom,
+        onFinished: submit,
+    });
+
+    const handleBackButtonPress = () => {
+        // TODO: connect to the fist step of the wallet setup https://github.com/Expensify/App/issues/36648
```

Comment:
```suggestion
        if (isEditing) {
          goToTheLastStep();
          return;
        }
        // TODO: connect to the fist step of the wallet setup https://github.com/Expensify/App/issues/36648
```

In case of `isEditing` flow, back button should go back to the confirmation screen. 

Bug:

https://github.com/Expensify/App/assets/14358475/b17e6772-733d-4d9f-90a4-6151d2b54662

Reply:
@koko57 has this been addressed?

Reply:
@mountiny yes it was addressed